src/coreclr/vm/gcenv.ee.cpp

```diff
@@ -1094,8 +1094,8 @@ void GCToEEInterface::HandleFatalError(unsigned int exitCode)
 bool GCToEEInterface::EagerFinalized(Object* obj)
 {
     MethodTable* pMT = obj->GetGCSafeMethodTable();
-    if (pMT == pWeakReferenceMT ||
-        pMT->GetCanonicalMethodTable() == pWeakReferenceOfTCanonMT)
+    if (pMT == g_pWeakReferenceClass ||
+        pMT->HasSameTypeDefAs(g_pWeakReferenceOfTClass))
```

Comment:
Delete `pWeakReferenceOfTCanonMT`  in gcenv.ee.static.cpp and gcenv.ee.standalone.cpp as well?